classes/citation/pid/BasePid.php

```diff
@@ -98,46 +87,17 @@ public static function normalize(?string $string): string
         /* @var BasePid $class */
         $class = get_called_class();
 
-        // no prefix defined, return original string
-        if (empty($class::prefix)) {
-            return $string;
-        }
-
-        $prefixInCorrect = $class::prefixInCorrect;
-
-        // prefix without https://
-        $prefixAlt = str_ireplace('https://', '', $class::prefix);
-
-        // make secure
-        $string = str_ireplace('http://', 'https://', $string);
-
-        // process longer first, e.g. dx.doi.org before doi.org
-        usort($prefixInCorrect, function ($a, $b) {
-            return strlen($b) - strlen($a);
-        });
-
-        // common mistakes, e.g. doi.org:10.123/tib123
-        $fixes = [
-            $prefixInCorrect,
-            "{$prefixAlt}: ",
-            "{$prefixAlt}:",
-            "{$prefixAlt} ",
-            "www.{$prefixAlt}"
-        ];
-        $string = str_ireplace($fixes, $prefixAlt, $string);
-
-        // add https://
-        $string = str_replace($prefixAlt, "https://{$prefixAlt}/", $string);
-
-        // clean doubles
-        $doubles = [
-            "https://{$prefixAlt}//",
-            "https://https://{$prefixAlt}/",
-            "https://https://{$prefixAlt}//"
-        ];
-        $string = str_ireplace($doubles, "https://{$prefixAlt}/", $string);
-
-        return trim($string, $class::defaultTrimCharacters);
+        $prefixes = array_merge(
+            [$class::prefix, $class::prefix . ' '],
+            [$class::urlPrefix],
+            $class::alternatePrefixes,
+            array_map(fn($value) => trim($value) . ' ', $class::alternatePrefixes)
+        );
+        $prefixes = array_filter($prefixes, fn($value) => !empty(trim($value)));
+        $prefixes = array_unique($prefixes);
+        usort($prefixes, fn($a, $b) => strlen($b) - strlen($a));
+
+        return trim(str_ireplace($prefixes, '', $string), $class::defaultTrimCharacters);
```

Comment:
why do we trim these default characters `.` and `/` ?
Would this DOI 10.123/tib123 then look like this 10123tib123 ?